fdbclient/VersionVector.h

```diff
@@ -144,7 +146,59 @@ struct VersionVector {
 
 	template <class Ar>
 	void serialize(Ar& ar) {
-		serializer(ar, versions, maxVersion);
+		// Encoded version of the "version vector":
+		//
+		// Encode localities by using RLE: Capture localities and their run lengths.
+		// Note that the order the localities appear in this map matches the order that they appear
+		// in the version vector.
+		std::map<int8_t, int8_t> tagLocalities;
+		// Capture the ids of tags (while preserving the order that they appear in the version vector).
+		// @todo We may be able to minimize the number of bytes used to represent the tag ids.
+		std::vector<uint16_t> tagIds;
```

Comment:
I think we can copy/read the tag ids and the tag commit versions directly into the serialization buffer, thereby avoiding the need to create these vectors. Working on it.